tests/test_validators.py

```diff
@@ -235,3 +235,12 @@ def test_invalid_uri_with_invalid_path(invalid_uri):
         validators.Validator().check_validity_of(
             'host', 'path',
         ).validate(uri)
+
+
+def test_rfc_4007_ipv6_zone_ids_invalid_host():
```

Comment:
Would you be willing to add a test case that shows that normalizing this same URIRef passes validation?

Reply:
Done!